src/components/app-field.js

```diff
@@ -1,10 +1,86 @@
 import React from 'react'
 import { connect } from 'react-redux'
-import { Table, Icon, Popup } from 'semantic-ui-react'
+import { Table, Icon, Popup, Button, Input } from 'semantic-ui-react'
+import axios from 'axios'
+
+import { getCookie } from 'formula_one'
+
+import { urlAppHiddenDetail } from '../urls'
 
 import main from '../css/app-field.css'
 
 class AppField extends React.PureComponent {
+  constructor(props) {
+    super(props)
+    this.state = {
+      password_verified: this.props.password_verified,
+      formOpen: false,
+      password: '',
+      hiddenData: '',
+      passwordPlaceholder: 'Enter password',
+      passwordError: false,
+    }
+  }
+
+  handleSubmitError = (data) => {
+    this.setState({
+      password: '',
+      passwordPlaceholder: data,
+      passwordError: true,
+    })
+  }
+
+  handleOpen = () => this.setState({ formOpen: true })
+
+  handleClose = () =>
+    this.setState({
+      formOpen: false,
+      passwordError: false,
+      password: '',
+      passwordPlaceholder: 'Enter password',
+    })
+
+  handlePasswordChange = (e, { value }) => {
+    this.setState({
+      password: value,
+    })
+  }
+
+  handleSubmitSecret = () => {
+    if(this.state.password === ''){
+      this.handleSubmitError("Enter something")
+      return
+    }
+
+    const headers = {
+      "X-CSRFToken": getCookie("csrftoken"),
+      "content-type": "application/json",
+    }
+
+    axios
+      .post(
+        urlAppHiddenDetail(),
+        {
+          password: this.state.password,
+          id: this.props.activeApp.data["id"],
+        },
+        { headers: headers }
+      )
+      .then((res) => {
+        this.setState({
+          password_verified: true,
+          hiddenData: res.data[this.props.field],
+        })
+      })
+      .catch((err) => this.handleSubmitError(err.response.data))
+  }
+
+  handleKeyDown = (event) => {
+    if(event.key === 'Enter'){
```

Comment:
It is better to use `event.keyCode === <code for Enter>`. Reference: https://www.w3schools.com/jsref/event_key_keycode.asp